app/src/main/java/com/infomaniak/drive/ui/fileList/SearchFiltersFragment.kt

```diff
@@ -0,0 +1,192 @@
+import androidx.core.os.bundleOf
+import androidx.core.view.isVisible
+import androidx.fragment.app.Fragment
+import androidx.navigation.fragment.findNavController
+import androidx.navigation.fragment.navArgs
+import androidx.navigation.navGraphViewModels
+import com.google.android.material.card.MaterialCardView
+import com.infomaniak.drive.R
+import com.infomaniak.drive.data.cache.DriveInfosController
+import com.infomaniak.drive.data.models.ConvertedType
+import com.infomaniak.drive.data.models.SearchCategoriesOwnershipFilter
+import com.infomaniak.drive.ui.fileList.fileDetails.SelectCategoriesFragment
+import com.infomaniak.drive.utils.getBackNavigationResult
+import com.infomaniak.drive.utils.safeNavigate
+import com.infomaniak.drive.utils.setBackNavigationResult
+import com.infomaniak.lib.core.utils.toPx
+import kotlinx.android.synthetic.main.fragment_search_filters.*
+
+class SearchFiltersFragment : Fragment() {
+
+    private val searchFiltersViewModel: SearchFiltersViewModel by navGraphViewModels(R.id.searchFiltersFragment)
+    private val navigationArgs: SearchFiltersFragmentArgs by navArgs()
+
+    private val categoryRights = DriveInfosController.getCategoryRights()
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View =
+        inflater.inflate(R.layout.fragment_search_filters, container, false)
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initializeFilters()
+        handleCategoryRights()
+        setToolbar()
+        setDateAndTypeFilters()
+        setCategoriesOwnershipFilters()
+        setClearButton()
+        setSaveButton()
+        listenToFiltersUpdates()
+    }
+
+    private fun initializeFilters() = with(searchFiltersViewModel) {
+        if (date.value == null) date.value = navigationArgs.date
+        if (type.value == null) type.value = navigationArgs.type?.let(ConvertedType::valueOf)
+        if (categories == null) {
+            categories = navigationArgs.categories?.toTypedArray()?.let(DriveInfosController::getCurrentDriveCategoriesFromIds)
+        }
+        if (categoriesOwnership == null) categoriesOwnership = navigationArgs.categoriesOwnership
+
+        updateAllFiltersUI()
+    }
+
+    private fun handleCategoryRights() {
+        val isVisible = categoryRights?.canReadCategoryOnFile == true
+        categoriesTitle.isVisible = isVisible
+        chooseCategoriesFilter.isVisible = isVisible
+    }
+
+    private fun setToolbar() {
+        toolbar.setNavigationOnClickListener { findNavController().popBackStack() }
+    }
+
+    private fun setDateAndTypeFilters() = with(searchFiltersViewModel) {
+        dateFilter.setOnClickListener { safeNavigate(R.id.searchFilterDateDialog, bundleOf("date" to date.value)) }
+        typeFilter.setOnClickListener { safeNavigate(R.id.searchFilterTypeDialog, bundleOf("type" to type.value)) }
```

Comment:
I don't think that in terms of maintainability using raw keys is a good thing, `SearchFilterDateBottomSheetDialogArgs` and `SearchFilterTypeBottomSheetDialogArgs` it'll be a good idea i think